File: solvingProblemAgent/c++/graph/state.cpp
#ifndef STATE_CPP
#define STATE_CPP

#include "transition.cpp"
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class transition;

class state{
    private:
        string cityName;
        vector<transition*> adjacentList;
    public:
        state(string cityName){
            this->cityName = cityName;
        }
        string getCityName(){
            return this->cityName;
        } 
        void addAdjacentList(transition* _transition){
            this->adjacentList.push_back(_transition);
        }
        vector<transition*> getAdjacentList(){
            return this->adjacentList;
        }
        /* string toString(){
            this->adjacentList[0]->toString();
            return "z";
        } */
};

#endif

File: solvingProblemAgent/java/src/graph/State.java
package src.graph;

import java.util.ArrayList;

public class State {
    private String cityName;
    private ArrayList<Transition> adjacentList;

    public State(String cityName){
        this.adjacentList = new ArrayList<Transition>();
        this.cityName = cityName;
    }
    public String getCityName(){
        return this.cityName;
    }
    public ArrayList<Transition> getAdjacentList(){
        return this.adjacentList;
    }
    public String toString(){
        String string = "State: "+this.cityName+"\n";
        for(Transition currentTransition: this.adjacentList){
            string+=currentTransition.toString()+"\n";
        }
        return string;
    }
}
